ckanext/saml2/plugin.py

```diff
@@ -353,16 +389,22 @@ def _create_or_update_user(self, user_name, saml_info):
         context = {'schema': user_schema, 'ignore_auth': True}
 
         if is_new_user:
+            user_name = _get_random_username_from_email(saml_info['email'][0])
+            gen = saml_info['id'][0]
+            data_dict['id'] = unicode(uuid.uuid4())
+            data_dict['name'] = user_name
```

Comment:
What about `user_name` that passed into this function? As i understand, we define whether user should be updated or deleted depending on that value. If you generate another name, than you end up in second attempt to create the same user? So, why you do not account anyhow this param?